.github/workflows/workflow-autoapply.yml

```diff
@@ -0,0 +1,183 @@
+name: Terraform
+
+on:
+  workflow_call:
+    inputs:
+      comment_plan:
+        description: Create a pull request comment of the plan in addition to adding it to the summary page
+        required: false
+        type: boolean
+        default: true
+
+      config:
+        description: A JSON (with comments) config string
+        required: true
+        type: string
+
+      default_branch:
+        required: false
+        type: string
+        default: main
+
+      refresh_on_pr:
+        required: false
+        type: boolean
+        default: true
+
+      terraform_version:
+        required: true
+        type: string
+
+      apply_timeout:
+        required: false
+        default: 60
+        type: number
+
+      artifacts_url:
+        description: Download an archive from the specified url and extract it under .artifacts in the root of the workspace
+        required: false
+        type: string
+        default: ""
+
+    outputs:
+      had_changes:
+        value: ${{ toJson(jobs.plan.outputs.*) != '[]' }}
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+
+jobs:
+  plan_apply:
+    name: Plan and Apply
+    runs-on: [self-hosted, Linux, AWS]
+    timeout-minutes: 60
+
+    environment: ${{ github.event_name != 'pull_request' && matrix.environment || null }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: ${{ fromJson(inputs.config).*.workspaces.*.environment }}
+
+    steps:
+    - id: parse
+      uses: Brightspace/terraform-workflows/actions/parse-config@v4
+      with:
+        # GHA's fromJson() supports comments in JSON strings which is nice from
+        # an end-user perspective (e.g. naming account ids). toJson(fromJson())
+        # is used in order to strip these comments out before parsing it ourselves
+        config: ${{ toJson(fromJson(inputs.config)) }}
+        max_environments: 50
+
+    - uses: Brightspace/third-party-actions@actions/checkout
+
+    - if: ${{ github.event_name != 'pull_request' }}
+      name: 'Assert: Branch is up to date'
+      uses: Brightspace/assert-git-remote-ref-action@main
+      with:
+        remote: origin
+        ref: ${{ inputs.default_branch }}
+
+    - if: ${{ inputs.artifacts_url != '' }}
+      name: 'Extract artifacts'
+      uses: Brightspace/s3-artifact-actions/extract@master
+      with:
+        path: ${{ fromJson(steps.parse.outputs.config)[matrix.environment].workspace_path }}/.artifacts
+        artifacts_url: ${{ inputs.artifacts_url }}
+
+    - id: plan
+      uses: Brightspace/terraform-workflows/actions/plan@v4
+      with:
+        comment_plan: ${{ inputs.comment_plan }}
+        config: ${{ toJson(fromJson(steps.parse.outputs.config)[matrix.environment]) }}
+        terraform_version: ${{ inputs.terraform_version }}
+        refresh_on_pr: ${{ inputs.refresh_on_pr }}
+        skip_archive: true
```

Comment:
We _may_ still want to upload the plans to s3 for historical/auditing purposes, but I think this is a fine start